### DISCUSSION TITLE
🗺️ Tell us more about your project

### DISCUSSION THREAD

**Discussion by aycabas** [microsoft/hack-together#33] [Show and tell]
We are excited to hear more about your project! 👏 

If you are struggling to find a project idea, we've got you! There is a list of **[📃 Top Microsoft Graph Scenarios](https://github.com/microsoft/hack-together/blob/main/top-scenarios.md)** available! You may build one of these scenarios to win [🏆 exciting prizes](https://github.com/microsoft/hack-together#prizes-), or get inspired and build your own project ideas!

Don't forget to submit your project until March 15th! You may submit your project here when it's ready: **[🚀 Project Submission](https://github.com/microsoft/hack-together/issues/new?assignees=&labels=&template=project.yml&title=Project%3A+%3Cshort+description%3E)**

Check out this video for step by step project submission guidance:

<https://user-images.githubusercontent.com/11164679/217246172-0aaea858-a4be-4fb8-abf2-beebfc1544ef.mp4>

**Comment by jai-dewani** [comment]
Hey @aycabas, I think the link to *Top Microsoft Graph Scenarios* is pointing to the wrong URL -> `https://github.com/top-scenarios.md`  
It should have been `https://github.com/microsoft/hack-together/blob/main/top-scenarios.md`

**Reply by aycabas** [reply]
Thanks @jai-dewani, I updated the link 🙏

**Comment by IRobot1** [comment]
Component to browse a user's OneDrive folder from within a VR application.  Filter allows user to select different models formats to preview in the VR scene.  Support selecting textures for apply to current model.

**Reply by aycabas** [reply]
That's an interesting one @IRobot1, looking forward to seeing the outcomes 👏

**Reply by waldekmastykarz** [reply]
🤩

**Comment by aunjum** [comment]
Hi I am Tanvir, My project is MeetBros, a meeting scheduling app that helps users find the perfect time and place to meet with friends, colleagues, or business partners. It uses the Microsoft Graph API to access the user's calendar and suggest available times for meetings based on the availability of all participants. The app also uses MAUI to provide a cross-platform user interface that works seamlessly on all devices.

**Reply by waldekmastykarz** [reply]
Awesome idea @aunjum! Looking forward to seeing your hack 👏

**Comment by PabloLION** [comment]
Hi I'm Pablo. I'm new to Microsoft 365.  Can we use [Microsoft Bot Framework](https://dev.botframework.com/) or [Azure Bot Service](https://azure.microsoft.com/en-us/products/bot-services/) to train some chatbot with the data from Microsoft Graph API or Microsoft Graph Data Connect?

**Reply by waldekmastykarz** [reply]
Absolutely!

**Reply by aycabas** [reply]
@PabloLION, welcome to the hack! You can absolutely use Bot Framework or Azure Bot Service with Microsoft Graph .NET SDK. Make sure to create your bot using .NET, so you can consume the right SDK. Keep us posted if you have any other questions 😊

**Comment by KamilSiebyla** [comment]
Hi all, 

when it comes to my project - I am creating application (for this event needs I am using .NET/Blazor) which will allow me to see instantly if person I am looking will be available soon - and if someone will be available I will show time slots when there is possibility to reach out with option to quick message on Teams/E-mail with something simple like "Hi, do you have a minute?". I will try to add some more features if there will be enough time of course. In general I just wanted to have something usable and working at the end.

Why something like this you could ask :) So, as for now I am usually checking folks calendars - but I wanted to make it little easier. 

My plan after this hack is to incorporate this feature to teams in the end to actually use it on daily basis.

This is all from me - not very innovative but will be useful I believe :) 

Thanks and have nice evening/day :) 

/Kamil

**Reply by waldekmastykarz** [reply]
Cool idea! Looking forward to seeing your hack!

**Comment by kevin-gatimu** [comment]
Hello everyone, my name is Kevin Comba Gatimu, am a Beta MLSA. I joined this hackathon to learn. I have never developed any app with Blazor or any dotnet library, because I struggle with OOP because of my background of functional programming using JS libraries, but I have done some console apps with C#. I am working on a on a simplified calendar app with this basic features. 

  1. View all events on your calendar.
  2. Add events to my calendar.
  3. View a summary of events per day.
 
I hope we will have a good learning experience.

**Reply by waldekmastykarz** [reply]
Welcome @kelcho-spense! Great to have you with us!

**Comment by EmreAka** [comment]
I'm just going to make a Blazor server application for managing and displaying todos, calendar events, etc. I've got no experience with both Graph API or Azure Active Directory. It's going tough for me. Thanks to the templates you shared for us, I managed to fetch my Todos.

**Reply by asieradzk** [reply]
Epic 🚀🚀🚀

**Reply by EmreAka** [reply]
> Epic 🚀🚀🚀

It is your epicness. Very kind of you. Thank you, BlueKiwi.

**Reply by waldekmastykarz** [reply]
Cool! Looking forward to seeing your hack!

**Comment by asieradzk** [comment]
I aim to develop an application that utilizes ChatGPT to provide succinct summaries of emails and delivers these summaries through push notifications on Android mobile devices.

**Reply by EmreAka** [reply]
Sounds cool!

**Reply by waldekmastykarz** [reply]
Cool! Looking forward to seeing it!

**Comment by castlecreati** [comment]
Hello Everyone,
I am trying to respond to a hypothetical Training Institute whose teachers are external. Creating the class calendar is very complex and they solve it by designing them in Excel and exporting them to sharepoint lists where they then access them with a calendar view. The view does not support more than the subject taught. The security department is interested in getting a list of the teachers attending on a given day.
My idea is to create an app that accesses the information from these sharepoint lists and with the data from these calendars create assignments of teachers (obtained from a security group) to certain classes or sections, and the corresponding subject. Then get the daily attendance of teachers of those assignments.
At the time I want to try Blazor to make the app, I had never touched it and it seems very attractive. 
I doubt I'll have time :)

**Reply by waldekmastykarz** [reply]
Awesome idea @castlecreati! Crossing my fingers for you!

**Comment by kusscaroline** [comment]
So, 
I have just done a skeleton of my project and published to azure. I just ned to fill it in, but can't seem to decide on what I am building....
I chose the console app, which I am guessing may be the least time consuming - anyway? - help please?

Thank you

**Reply by waldekmastykarz** [reply]
Correct, a console app would be probably the easiest. Have you seen our [top scenarios](https://github.com/microsoft/hack-together/blob/main/top-scenarios.md)? These might give you some inspiration.

**Comment by loladeajayi** [comment]
Hello all, if I want to get users in my team only not the entire organization and I need their employee hire data, how do I go about that please? Thank you.

**Reply by waldekmastykarz** [reply]
You'd start by retrieving your manager:

```cs
var graphClient = new GraphServiceClient(requestAdapter);
var manager = await graphClient.Me.Manager.GetAsync();
```

then, you'd retrieve everyone who reports to the manager (user-id is the id of the manager retrieved in the previous call):

```cs
var peers = await graphClient.Users["{user-id}"].DirectReports.GetAsync();
```

From the list of peers you'd need to filter out yourself by comparing the IDs of the user with your own ID.

What kind of hire data would you like to retrieve? Here's the list of data available about users on Microsoft Graph: https://learn.microsoft.com/graph/api/resources/user?view=graph-rest-1.0#properties

**Reply by loladeajayi** [reply]
Thank you, found the employee hire date from the link you shared.

**Reply by loladeajayi** [reply]
For the sample data in the tenant created, there seem not to be manager data. It shows the error below:
"code": "Request_ResourceNotFound",
        "message": "Resource 'manager' does not exist or one of its queried reference-property objects are not present.

**Reply by loladeajayi** [reply]
> You'd start by retrieving your manager:
> 
> ```cs
> var graphClient = new GraphServiceClient(requestAdapter);
> var manager = await graphClient.Me.Manager.GetAsync();
> ```
> 
> then, you'd retrieve everyone who reports to the manager (user-id is the id of the manager retrieved in the previous call):
> 
> ```cs
> var peers = await graphClient.Users["{user-id}"].DirectReports.GetAsync();
> ```
> 
> From the list of peers you'd need to filter out yourself by comparing the IDs of the user with your own ID.
> 
> What kind of hire data would you like to retrieve? Here's the list of data available about users on Microsoft Graph: https://learn.microsoft.com/graph/api/resources/user?view=graph-rest-1.0#properties

Hello @waldekmastykarz @aycabas , please is there a way for me to update the tenant data to have manager information? I still have the  error below when I click on manager under the Tenant data when I login to test from this link https://developer.microsoft.com/en-us/graph/graph-explorer, https://graph.microsoft.com/v1.0/me/manager
"code": "Request_ResourceNotFound",
"message": "Resource 'manager' does not exist or one of its queried reference-property objects are not present.

Thank you.

**Reply by waldekmastykarz** [reply]
You can update manager data by navigating to the Azure Portal at https://portal.azure.com. In there, go to Azure Active Directory > Users, find your user account and in the properties update the manager property.

**Comment by kusscaroline** [comment]
Hi,

Please can anyone help?
Blazor Template: 
1. How do you extract a data from the Fetch Weather Data
2. How do you then @bind that data to something?
3. How you then send out a public message to a group (Teams or 365)?
Thank you

**Reply by aycabas** [reply]
Hi @kusscaroline, you can use any weather data api you prefer and make an HTTP call to retrieve the data from the weather api. 
To send an email, please check out this doc: https://learn.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=http
To send a chat message to a group or user, please check out this doc: https://learn.microsoft.com/en-us/graph/api/chatmessage-post?view=graph-rest-1.0&tabs=http

Let me know if you need further assistance.

**Comment by bharat1203** [comment]
Hi everyone,
I have developed a application using Asp.net to authenticate the user with secure login to Microsoft office by creating the Microsoft Identity platform.
So that it will allow only the authenticated users to access the information.
After successful authenticate we schedule the events in calendars.

**Reply by waldekmastykarz** [reply]
Cool!

**Comment by vaishnavdev** [comment]
HI everyone, my name is P. Vaishnav, I have developed a Meeting App using which the users can,

1. Login to your application using Identity platform for Authentication and Authorization
2. Create an Event / Meeting in their Outlook calendar
3. Edit / Update an Event / Meeting in the outlook calendar
4. Delete an Event / Meeting from the Outlook calendar
5. View all Events / Meetings in the Outlook calendar

Thank you for this oppurtunity. i look forward for more challenges and add more features to this application.

**Reply by waldekmastykarz** [reply]
Awesome! Thank you for sharing!

**Comment by kusscaroline** [comment]
So, I just got fed up...and submitted my App :)
I was having so much hard time with the logging of members in my Tenants...blah...blah.... 👎 
And that Blazor's Layout??! - so rigid, it took my 2days for not much 👎  
Good luck, everyone! - Now, off to make the dinner 👍
